crates/shared/src/price_estimation/native/coingecko.rs

```diff
@@ -55,11 +55,11 @@ impl CoinGecko {
 impl NativePriceEstimating for CoinGecko {
     fn estimate_native_price(&self, token: Token) -> BoxFuture<'_, NativePriceEstimateResult> {
         async move {
-            let url = format!(
-                "{}/{}?contract_addresses={token:#x}&vs_currencies=eth",
-                self.base_url, self.chain
-            );
-            let mut builder = self.client.get(&url);
+            let mut url = crate::url::join(&self.base_url, &self.chain);
+            url.query_pairs_mut()
+                .append_pair("contract_addresses", &token.to_string())
```

Comment:
@squadgazzz this is incorrect...

Reply:
🙈 is this because `token.to_string()` doesn't encode properly? We should be better at testing our changes before merging (I believe you even added a test case that would have caught this)